back/app/services/print_custom_fields_service.rb

```diff
@@ -0,0 +1,278 @@
+# frozen_string_literal: true
+
+require 'prawn'
+require 'prawn/measurement_extensions'
+
+class PrintCustomFieldsService
+  attr_reader :participation_context, :custom_fields, :params, :previous_cursor
+
+  # QUESTION_TYPES = %w[select multiselect text text_multiloc multiline_text html_multiloc linear_scale]
+  # Disable linear scales for now, as they're not detected correctly by form parser
+  QUESTION_TYPES = %w[select multiselect text text_multiloc multiline_text html_multiloc]
+  FORBIDDEN_HTML_TAGS_REGEX = %r{</?(div|span|ul|ol|li|em|img|a){1}[^>]*/?>}
+
+  def initialize(participation_context, custom_fields, params)
+    @participation_context = participation_context
+    @custom_fields = custom_fields
+    @params = params
+    @previous_cursor = nil
+  end
+
+  def create_pdf
+    pdf = Prawn::Document.new(page_size: 'A4', top_margin: 2.cm)
+
+    if params[:name] == 'true'
+      render_text_field_with_name(
+        pdf,
+        I18n.with_locale(locale) { I18n.t('form_builder.pdf_export.full_name') }
+      )
+    end
+
+    if params[:email] == 'true'
+      render_text_field_with_name(
+        pdf,
+        I18n.with_locale(locale) { I18n.t('form_builder.pdf_export.email_address') }
+      )
+    end
+
+    custom_fields.each_with_index do |custom_field, i|
+      field_type = custom_field.input_type
+
+      # If this is a survey, the first field will be a 'page'.
+      # Since the pdf is initialized with an empty page,
+      # we can skip this.
+      next if i == 0 && field_type == 'page'
+
+      if field_type == 'page'
+        pdf.start_new_page(size: 'A4')
+      end
+
+      # Skip unsupported question types
+      next unless QUESTION_TYPES.include? field_type
+
+      render_field(pdf, custom_field)
+    end
+
+    # Add page numbers
+    page_copy = I18n.with_locale(locale) { I18n.t('form_builder.pdf_export.page') }
+    page_number_format = "#{page_copy} <page>"
+    page_number_options = {
+      at: [pdf.bounds.right - 150, 275.mm],
+      width: 150,
+      align: :right
+
+    }
+
+    pdf.number_pages page_number_format, page_number_options
+
+    pdf
+  end
+
+  private
+
+  def render_text_field_with_name(pdf, name)
+    title_multiloc = {}
+    title_multiloc[locale] = name
+
+    text_field = CustomField.new({
+      input_type: 'text',
+      title_multiloc: title_multiloc
+    })
+
+    render_field(pdf, text_field)
+  end
+
+  def render_field(pdf, custom_field)
+    field_type = custom_field.input_type
+
+    # The .group block makes sure that everything
+    # inside of it will be on a new page if there
+    # is not enough space on the current page
+    pdf.group do |pdf|
```

Comment:
@luuc The `pdf` variable inside this loop (or the one outside) needs to be changed as it has the same name as the outer variable and is causing a rubocop issue